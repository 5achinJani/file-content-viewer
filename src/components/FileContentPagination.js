import React from "react";

import { Pagination, PaginationItem, PaginationLink } from "reactstrap";
import FontAwesome from "react-fontawesome";

export const FileContentPagination = ({ page, totalPage, onPaginate }) => {
  if (!totalPage) {
    return null;
  }
  const prevNext = number => {
    if (number < 1 || number > totalPage) {
      return false;
    } else if (page === number) {
      return false;
    }
    onPaginate(number);
  };
  return (
    <React.Fragment>
      <Pagination className="justify-content-center">
        <PaginationItem
          disabled={page === 1}
          title="Go to beginner of file"
          onClick={() => {
            onPaginate(1);
          }}
        >
          <PaginationLink>
            <FontAwesome name="angle-double-left" />
          </PaginationLink>
        </PaginationItem>
        <PaginationItem
          disabled={page - 1 < 1}
          title="Previous 10 lines"
          onClick={() => {
            prevNext(page - 1);
          }}
        >
          <PaginationLink>
            <FontAwesome name="angle-left" />
          </PaginationLink>
        </PaginationItem>

        <PaginationItem
          disabled={page + 1 > totalPage}
          title="Next 10 lines"
          onClick={() => {
            prevNext(page + 1);
          }}
        >
          <PaginationLink>
            <FontAwesome name="angle-right" />
          </PaginationLink>
        </PaginationItem>

        <PaginationItem
          disabled={page === totalPage}
          title="Go to end of file"
          onClick={() => {
            onPaginate(totalPage);
          }}
        >
          <PaginationLink>
            <FontAwesome name="angle-double-right" />
          </PaginationLink>
        </PaginationItem>
      </Pagination>
    </React.Fragment>
  );
};
