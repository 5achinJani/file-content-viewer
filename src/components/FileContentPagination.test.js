import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { FileContentPagination } from "./FileContentPagination";

it("should render pagination", () => {
  const props = {
    page: 2,
    totalPage: 5,
    onPaginate: jest.fn()
  };
  const component = renderer.create(<FileContentPagination {...props} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("should not render pagination", () => {
  const component = renderer.create(<FileContentPagination />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("should paginate: next & end", () => {
  const props = {
    page: 1,
    totalPage: 3,
    onPaginate: jest.fn()
  };
  const component = shallow(<FileContentPagination {...props} />);
  const nextBtn = component.find('[name="js-item-next"]');
  const exec = (btn) => {
    expect(btn.props().disabled).toBeFalsy();
    btn.simulate("click");
    expect(props.onPaginate).toHaveBeenCalled();
  }
  exec(nextBtn);
});

it("should not paginate: next", () => {
  const props = {
    page: 1,
    totalPage: 1,
    onPaginate: jest.fn()
  };
  const component = shallow(<FileContentPagination {...props} />);
  const nextBtn = component.find('[name="js-item-next"]');
  expect(nextBtn.props().disabled).toBeTruthy();
  nextBtn.simulate("click");
  expect(props.onPaginate).not.toHaveBeenCalled();
});

it("should paginate: back", () => {
  const props = {
    page: 2,
    totalPage: 3,
    onPaginate: jest.fn()
  };
  const component = shallow(<FileContentPagination {...props} />);
  const backBtn = component.find('[name="js-item-back"]');
  expect(backBtn.props().disabled).toBeFalsy();
  backBtn.simulate("click");
  expect(props.onPaginate).toHaveBeenCalled();
});

