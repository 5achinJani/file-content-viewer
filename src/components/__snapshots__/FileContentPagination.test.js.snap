// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render pagination 1`] = `null`;

exports[`should render pagination 1`] = `
<ul
  className="justify-content-center pagination"
>
  <li
    className="page-item"
    name="js-item-start"
    onClick={[Function]}
    title="Go to beginner of file"
  >
    <a
      aria-label={undefined}
      className="page-link"
    >
      <span
        aria-hidden={true}
        className="fa fa-angle-double-left"
      />
    </a>
  </li>
  <li
    className="page-item"
    name="js-item-back"
    onClick={[Function]}
    title="Previous 10 lines"
  >
    <a
      aria-label={undefined}
      className="page-link"
    >
      <span
        aria-hidden={true}
        className="fa fa-angle-left"
      />
    </a>
  </li>
  <li
    className="page-item"
    name="js-item-next"
    onClick={[Function]}
    title="Next 10 lines"
  >
    <a
      aria-label={undefined}
      className="page-link"
    >
      <span
        aria-hidden={true}
        className="fa fa-angle-right"
      />
    </a>
  </li>
  <li
    className="page-item"
    name="js-item-end"
    onClick={[Function]}
    title="Go to end of file"
  >
    <a
      aria-label={undefined}
      className="page-link"
    >
      <span
        aria-hidden={true}
        className="fa fa-angle-double-right"
      />
    </a>
  </li>
</ul>
`;
